### PR TITLE
Build: Add an option BUILD_NATIVE_ARCH

### PR DESCRIPTION
**Pull request by manticore-projects**
Adds a CMake Option `BUILD_NATIVE_ARCH`, which:
- sets CXXFLAGS: `-O3 -march=native -frecord-gcc-switches`
- allows for: `readelf -p .GCC.command.line build/release/duckdb`

## Example
```shell
make -j 16 BUILD_NATIVE_ARCH=1
readelf -p .GCC.command.line build/release/duckdb
```

### Output
```text
String dump of section '.GCC.command.line':
  [   451]  GNU C++11 12.2.1 20230201 -march=znver3 -mmmx -mpopcnt -msse -msse2 -msse3 -mssse3 -msse4.1 -msse4.2 -mavx -mavx2 -msse4a -mno-fma4 -mno-xop -mfma -mno-avx512f -mbmi -mbmi2 -maes -mpclmul -mno-avx512vl -mno-avx512bw -mno-avx512dq -mno-avx512cd -mno-avx512er -mno-avx512pf -mno-avx512vbmi -mno-avx512ifma -mno-avx5124vnniw -mno-avx5124fmaps -mno-avx512vpopcntdq -mno-avx512vbmi2 -mno-gfni -mvpclmulqdq -mno-avx512vnni -mno-avx512bitalg -mno-avx512bf16 -mno-avx512vp2intersect -mno-3dnow -madx -mabm -mno-cldemote -mclflushopt -mclwb -mclzero -mcx16 -mno-enqcmd -mf16c -mfsgsbase -mfxsr -mno-hle -msahf -mno-lwp -mlzcnt -mmovbe -mno-movdir64b -mno-movdiri -mmwaitx -mno-pconfig -mpku -mno-prefetchwt1 -mprfchw -mno-ptwrite -mrdpid -mrdrnd -mrdseed -mno-rtm -mno-serialize -mno-sgx -msha -mshstk -mno-tbm -mno-tsxldtrk -mvaes -mno-waitpkg -mwbnoinvd -mxsave -mxsavec -mxsaveopt -mxsaves -mno-amx-tile -mno-amx-int8 -mno-amx-bf16 -mno-uintr -mno-hreset -mno-kl -mno-widekl -mno-avxvnni -mno-avx512fp16 --param=l1-cache-size=32 --param=l1-cache-line-size=64 --param=l2-cache-size=512 -mtune=znver3 -O3 -O3 -O3 -std=c++11 -fPIE
```

### Objective
The goal is to get the fastest possible binary for my Epyc servers and performance tests.

### Cons
Of course the binaries would not be compatible with other architectures. Also the build time slightly increases.

### CODE REVIEW

**Comment by Mytherin** [comment]
The non-interpreted tests (i.e. the tests that are not file paths) are generally data ingestion/append/data modification tests. I would run the interpreted benchmarks as well to reach a more complete conclusion.

**Comment by manticore-projects** [comment]
> The non-interpreted tests (i.e. the tests that are not file paths) are generally data ingestion/append/data modification tests. I would run the interpreted benchmarks as well to reach a more complete conclusion.

Thank you, I am on it.
I was mostly surprised that the `-march-native` flags reduce the volatility.

File: Makefile [hunk 1]
```diff
@@ -170,6 +171,9 @@ endif
 ifeq (${DEBUG_STACKTRACE}, 1)
 	EXTENSIONS:=${EXTENSIONS} -DDEBUG_STACKTRACE=1
 endif
+ifeq (${BUILD_NATIVE_ARCH}, 1)
+	BUILD_NATIVE_ARCH_FLAG=-DBUILD_NATIVE_ARCH=1
```
[the remark this answers is not in the document]
Comment (manticore-projects) [reply Makefile:175]: Yes, sure. I am totally new here and still need to figure the preferences. I will amend that PR of course.
Thank you for the feedback!

**Comment by Mytherin** [comment]
Closing as this has been open for a while with no activity and the underlying files have changed significantly in the meantime

**Pull request closed without merging**